ipynb/AG_stool_temporal dynamics.ipynb

```diff
@@ -0,0 +1,664 @@
+     "text": [
+      "565\n",
+      "1553\n",
+      "1553\n"
+     ]
+    }
+   ],
+   "source": [
+    "unique_ids = []\n",
+    "unique_sampleA = []\n",
+    "unique_sampleB = []\n",
+    "\n",
+    "for i in deltas['HOST_SUBJECT_ID']:\n",
+    "    if i not in unique_ids:\n",
+    "        unique_ids.append(i)\n",
+    "        \n",
+    "for i in deltas['SampleID-A']:\n",
+    "    if i not in unique_sampleA:\n",
+    "        unique_sampleA.append(i)\n",
+    "        \n",
+    "for i in deltas['SampleID-B']:\n",
+    "    if i not in unique_sampleB:\n",
+    "        unique_sampleB.append(i)\n",
+    "\n",
+    "print(len(unique_ids))\n",
+    "print(len(unique_sampleA))\n",
+    "print(len(unique_sampleB))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "distance = []\n",
+    "for dl in range(len(absdeltas)):\n",
+    "    distance.append(dm[absdeltas['SampleID-A'][dl], absdeltas['SampleID-B'][dl]])\n",
+    "\n",
+    "absdeltas['Distance'] = distance"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "delmonths = []\n",
+    "for dl in range(len(absdeltas)):\n",
+    "    if absdeltas['DeltaInDays'][dl]/30 <= 1:\n",
+    "        delmonths.append(1)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 2:\n",
+    "        delmonths.append(2)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 3:\n",
+    "        delmonths.append(3)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 4:\n",
+    "        delmonths.append(4)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 5:\n",
+    "        delmonths.append(5)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 6:\n",
+    "        delmonths.append(6)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 7:\n",
+    "        delmonths.append(7)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 8:\n",
+    "        delmonths.append(8)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 9:\n",
+    "        delmonths.append(9)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 10:\n",
+    "        delmonths.append(10)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 11:\n",
+    "        delmonths.append(11)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 <= 12:\n",
+    "        delmonths.append(12)\n",
+    "    elif absdeltas['DeltaInDays'][dl]/30 > 12:\n",
+    "        delmonths.append(13)\n",
+    "\n",
+    "absdeltas['DeltaInMonths'] = delmonths"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "uni_sub_del = pd.DataFrame()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "host_1mo = {}\n",
```

Comment:
It seems like this entire cell could be simplified with another layer of iteration, otherwise it seems like this is very prone to errors. Could you revise this and if you need help let me know?